### DISCUSSION TITLE
Requesting an automatic update method for Teslamate

### DISCUSSION THREAD

**Discussion by mikec-design** [teslamate-org/teslamate#2484] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Each time I update Teslamate I find that it completely breaks the application. I then have to spend hours, days, or longer to diagnose the problems and test solutions.

I've repeatedly been caught off guard by updates to the config file for new versions. The need to check review the config file isn't mentioned in the "Upgrading to a new version" documentation.

A friendlier method comparable to PiHole's "pihole -up" would be incredibly helpful.

### Expected Behavior

Updates to Teslamate should progress smoothly, updating any necessary components or config files as needed with a minimum of manual editing or configuration.

### Relevant log output

```shell
N/A
```

### Type of installation

Docker

### Version

v1.26.1

**Comment by cwanja** [comment]
If you have it installed through Docker, I am confused how `docker-compose pull` has broken anything? Unless you are referring to the recent issues where TeslaMate was updated to support Bullseye and it had some side effects. Otherwise, I have had zero issues of updating TeslaMate through docker-compose for nearly two years. Never once broke anything. 

`pihole -up` would not work since Pi-hole is installed as a service. And I do not see TeslaMate getting to that point. 

If you want, here is my update script. You can set it as a cron job and forget about it:
```
#!/bin/bash

#Create date variable and assign it a value
date=`date +%F`

#echo "$date"

#Begin backup process
echo "Beginning backup process"
docker-compose exec -T database pg_dump -U teslamate teslamate > ./"$date"_teslamate.bck
echo "TeslaMate backed up `pwd`/"$date"_teslaback.bck"

#Begin TeslaMate update process
echo "Beginning TeslaMate update. docker-compose down, pull, forced prune and restart"\n
docker-compose down
docker-compose pull
docker system prune -f
docker-compose up -d
echo "TeslaMate updated!"
```

**Comment by BabyYeggie** [comment]
You can use Ouroboros, but change interval to 604800 (1 week):
version: '3'
services:
  ouroboros:
    container_name: ouroboros
    hostname: ouroboros
    image: pyouroboros/ouroboros
    environment:
      - CLEANUP=true
      - INTERVAL=604800
      - LOG_LEVEL=info
      - SELF_UPDATE=true
      - IGNORE=mongo influxdb postgres mariadb
      - TZ=America/Chicago
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

https://github.com/pyouroboros/ouroboros

**Reply by DrogoNevets** [reply]
Ouroboros is now end of life - just thought I would mention here incase people just blindly started using it

**Comment by init6-git** [comment]
This is not an issue. Never had a problem upgrading and even the bullseye was a Pi OS issue rather than a Teslamate one. Are you expecting Adrian to test Teslamate on every conceivable version of hardware and OS? As Cwanja says above, docker gets rid of a lot of the hassle, but you are responsible for the OS. Perhaps Teslafi would be a better fit for you.

**Reply by mikec-design** [reply]
Obviously I don't expect that every platform be tested. But a Raspberry Pi 3 is a fairly common platform, and while your updates have gone smoothly, mine never have.

My suggestion was for a friendlier method that would handle changes in the config file and update the app in a single go. Allowing people who are comfortable with the command line, but not in-depth Linux users to run Teslamate without having to debug other issues.

**Comment by adriankumpf** [comment]
Since good hints and suggestions have already been posted, I'll move this thread to a discussion.
